src/prof.c

```diff
@@ -256,7 +256,10 @@ prof_backtrace(prof_bt_t *bt) {
 
 	cassert(config_prof);
 
+	tsd_t *tsd = tsd_fetch();
```

Comment:
Let’s pass tsd down to here?

Also we should add the pre and post reentrancy for all backtrace impls (add shared begin / finish functions) not just the libc one.